src/apis/impl/DataOwnerApiImpl.ts

```diff
@@ -71,24 +73,39 @@ export class DataOwnerApiImpl implements DataOwnerApi {
     keyPair?: { publicKey: string; privateKey: string }
   ): Promise<{ publicKey: string; privateKey: string }> {
     const dataOwnerId = this.getDataOwnerIdOf(user)
-
-    const { publicKey, privateKey } = keyPair ?? (await this._generateKeyPair())
-    const jwk = {
-      publicKey: spkiToJwk(hex2ua(publicKey)),
-      privateKey: pkcs8ToJwk(hex2ua(privateKey)),
-    }
-    await this.cryptoApi.cacheKeyPair(jwk)
-    await this.keyStorage.storeKeyPair(`${dataOwnerId}.${publicKey.slice(-32)}`, jwk)
-
     const dataOwner = await this.cryptoApi.getDataOwner(dataOwnerId).catch((e) => {
       throw this.errorHandler.createErrorFromAny(e)
     })
+
     if (dataOwner == null) {
       throw this.errorHandler.createErrorWithMessage(
         `The current user is not a data owner. You must be either a patient, a device or a healthcare professional to call this method.`
       )
     }
 
+    let jwks: { privateKey: JsonWebKey; publicKey: JsonWebKey } | undefined
+    if (keyPair === undefined && !overwriteExistingKeys && dataOwner.dataOwner.publicKey !== undefined) {
```

Comment:
To cover with tests (may be unit tests ones to simplify): 
- A dataOwner without any key calls `initCryptoFor(user, true)` and will receive a new keypair (and its dataOwner will be updated);
- A dataOwner with an existing keypair, present in the localStorage calls `initCryptoFor(user, true)` and will receive its existing keypair; 
- A dataOwner with an existing keypair, but not in the localStorage calls `initCryptoFor(user, true)` and will receive a new keypair (and its dataOwner will be updated);
- A dataOwner with an existing keypair, but not in the localStorage calls `initCryptoFor(user)` and receives an error because we can't find any key in the keyStorage;
- A dataOwner with 2 existing keypairs, both present in the localStorage calls `initCryptoFor(user)` and receives its 2 existing keypairs; 

I'm also wondering: How should we manage the use cases where the user provided a keypair but has an existing one in the localStorage ? It should not really happen, I think we should throw an error in this case. 
I guess we should have clear tests as well about how this keypair argument should be used. For now, I'm a bit confused.